### PR TITLE
Images Interpolation with Stable Diffusion

### PR DESCRIPTION
**Pull request by AkiRusProd**
# What does this PR do?

This notebook shows several ways to interpolate images  using Stable Diffusion.  Image interpolation using Stable Diffusion is the process of creating intermediate images that smoothly transition from one given image to another, using a generative model based on diffusion. 

@MKhalusova

### CODE REVIEW

**Comment by MKhalusova** [comment]
Thank you for the contribution! Diffusion models are not my strong suit, so I'd like to ask @sayakpaul for a review.

File: notebooks/en/stable_diffusion_interpolation.ipynb [hunk 1]
```diff
@@ -0,0 +1 @@
...ter-widgets/base","_view_module_version":"1.2.0","_view_name":"LayoutView","align_content":null,"align_items":null,"align_self":null,"border":null,"bottom":null,"display":null,"flex":null,"flex_flow":null,"grid_area":null,"grid_auto_columns":null,"grid_auto_flow":null,"grid_auto_rows":null,"grid_column":null,"grid_gap":null,"grid_row":null,"grid_template_areas":null,"grid_template_columns":null,"grid_template_rows":null,"height":null,"justify_content":null,"justify_items":null,"left":null,"margin":null,"max_height":null,"max_width":null,"min_height":null,"min_width":null,"object_fit":null,"object_position":null,"order":null,"overflow":null,"overflow_x":null,"overflow_y":null,"padding":null,"right":null,"top":null,"visibility":null,"width":null}},"ffe65c700f0142df9b289a0e8b58ec65":{"model_module":"@jupyter-widgets/controls","model_module_version":"1.5.0","model_name":"DescriptionStyleModel","state":{"_model_module":"@jupyter-widgets/controls","_model_module_version":"1.5.0","_model_name":"DescriptionStyleModel","_view_count":null,"_view_module":"@jupyter-widgets/base","_view_module_version":"1.2.0","_view_name":"StyleView","description_width":""}}}}},"nbformat":4,"nbformat_minor":4}
```
Comment (sayakpaul) [inline notebooks/en/stable_diffusion_interpolation.ipynb:1]: <p>We need to split this cell up into multiple ones discussing each of the logical blocks. For example, you could consider grouping the code associated with embedding computation together and discussing it and then proceeding accordingly. Otherwise, it might be hard to follow for the readers. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/42/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

**Comment by sayakpaul** [comment]
Code wise, this look very nice to me ❤️ Thanks a lot for working on this. You could also consider discussing one/two use cases of image interpolation. 

I'd ask @MKhalusova to check the wording, phrases, etc.

File: notebooks/en/stable_diffusion_interpolation.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (AkiRusProd) [reply notebooks/en/stable_diffusion_interpolation.ipynb:1]: Hi! Thank you for your review. I opened a pull request to upload the files:
https://huggingface.co/datasets/huggingface/cookbook-images/discussions/7

**Comment by AkiRusProd** [comment]
Hello. I have made changes to the code as per @sayakpau's suggestions
 @MKhalusova

**Comment by AkiRusProd** [comment]
As for more use cases of image interpolation. I'll think about it.

**Comment by sayakpaul** [comment]
> As for more use cases of image interpolation. I'll think about it.

Just to be clear, I am not asking for including more examples. You have already shown four -- which is more than enough. I am asking you to think of some use cases that we could include at the beginning of the notebook that could further motivate the readers.

File: notebooks/en/_toctree.yml [hunk 2]
```diff
@@ -2,6 +2,10 @@
   sections:
   - local: index
     title: Open-Source AI Cookbook
+  - local: stable_diffusion_interpolation
+    title: Stable Diffusion Interpolation
+ -  local: tgi_messages_api_demo
+    title: Migrating from OpenAI to Open LLMs Using TGI's Messages API
```
Comment (sayakpaul) [inline notebooks/en/_toctree.yml:8]: Why are these changes being included here? Seems unrelated no?

File: notebooks/en/index.md [hunk 3]
```diff
@@ -7,6 +7,8 @@ applications and solving various machine learning tasks using open-source tools
 
 Check out the recently added notebooks: 
 
+- [Stable Diffusion Interpolation](stable_diffusion_interpolation)
+- [Migrating from OpenAI to Open LLMs Using TGI's Messages API](tgi_messages_api_demo)
```
Comment (sayakpaul) [inline notebooks/en/index.md:11]: Same.

Comment (MKhalusova) [reply notebooks/en/index.md:12]: I resolved the conflict

**Comment by AkiRusProd** [comment]
Hi, thanks for taking a look at this notebook. I have made changes to the code based on @sayakpau's suggestions (fixed links to gifs and added some different use cases).
@MKhalusova

File: notebooks/en/stable_diffusion_interpolation.ipynb (hunk 1 above)
Comment (MKhalusova) [reply notebooks/en/stable_diffusion_interpolation.ipynb]: <p>+1, this needs to be split a bit further to make it easier to follow. </p>

File: notebooks/en/stable_diffusion_interpolation.ipynb [hunk 4]
```diff
@@ -0,0 +1 @@
...
```
Comment (MKhalusova) [inline notebooks/en/stable_diffusion_interpolation.ipynb]: <p>"navigate through a high-dimensional latent space, which is a high-dimensional space." is a bit repetitive. maybe remove the part after the comma? </p><p> </p><p>It would be helpful to mention (just a sentence or two) some use cases for image interpolation. What kinds of real-world applications might use this? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/42/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (MKhalusova) [inline notebooks/en/stable_diffusion_interpolation.ipynb]: <p>While this works, let's replace <code>%pip</code> with <code>!pip</code>  for consistency with other notebooks. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/42/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (MKhalusova) [inline notebooks/en/stable_diffusion_interpolation.ipynb]: <p>Can you add a link to the model? (<a href="https://huggingface.co/runwayml/stable-diffusion-v1-5" target="_blank">https://huggingface.co/runwayml/stable-diffusion-v1-5</a>) and to the scheduler docs? (<a href="https://huggingface.co/docs/diffusers/en/api/schedulers/lms_discrete" target="_blank">https://huggingface.co/docs/diffusers/en/api/schedulers/lms_discrete</a>)</p><p>Can you also add a sentence or two explaining your choice of model?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/42/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (MKhalusova) [inline notebooks/en/stable_diffusion_interpolation.ipynb]: <p>It would be cool to explain your choice of the values for these parameters. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/42/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (MKhalusova) [inline notebooks/en/stable_diffusion_interpolation.ipynb]: <pre class="ql-syntax" spellcheck="false">Line #3.    # A negative prompt that can be used to steer the generation away from certain features; here, it is empty.
</pre><p>The comment says that the negative prompt is empty here, but it is not. Please update the comment</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/42/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (MKhalusova) [inline notebooks/en/stable_diffusion_interpolation.ipynb]: <p>Rephrasing suggestion:</p><p>"Finally, let's generate a series of images based on interpolated embeddings and then displaying these images. We'll iterate over an array of embeddings, using each to generate an image with specified characteristics like height, width, and other parameters relevant to image generation. Then we'll collect these images into a list. Once generation is complete we'll call the <code>display_image</code> function to save and display these images as GIF at a given save path."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/42/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (MKhalusova) [inline notebooks/en/stable_diffusion_interpolation.ipynb]: <p>I wouldn't call this "homework". Let's rephrase this part as "next steps" for example, and instead of list, can you rewrite this as a paragraph?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/42/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

File: notebooks/en/stable_diffusion_interpolation.ipynb (hunk 1 above)
Comment (AkiRusProd) [reply notebooks/en/stable_diffusion_interpolation.ipynb]: <p>Already split it</p>

**Comment by MKhalusova** [comment]
Thanks @sayakpaul for a review! 
I looked over the notebook - there are a few finishing touches to address, and then we should be good to merge. The PR with images is merged now too.

File: notebooks/en/stable_diffusion_interpolation.ipynb (hunk 4 above)
Comment (AkiRusProd) [reply notebooks/en/stable_diffusion_interpolation.ipynb]: <p>It seems there might be a misunderstanding, as I've already mentioned several use cases for image interpolation with Stable Diffusion in the provided introduction text.</p>

**Comment by AkiRusProd** [comment]
Hello, thanks for the review! But I have a question. In the introduction, I already covered three use cases for image interpolation using Stable Diffusion. I don't quite understand what exactly I should do. And I made the rest of the changes in accordance with your advice.
@MKhalusova

File: notebooks/en/stable_diffusion_interpolation.ipynb (hunk 4 above)
Comment (MKhalusova) [reply notebooks/en/stable_diffusion_interpolation.ipynb]: Fair enough

**Comment by AkiRusProd** [comment]
Hello, what are our next steps?
@MKhalusova

**Comment by MKhalusova** [comment]
Looks good to merge! But I will do social communications about this notebook tomorrow.

**Review by MKhalusova — APPROVED** [review]
